lib/autoproj/cli/ci.rb

```diff
@@ -149,41 +246,56 @@ def load_built_flags
                 path = @ws.build_report_path
                 return {} unless File.file?(path)
 
-                report = JSON.load(File.read(path))
-                report['build_report']['packages'].
-                    each_with_object({}) do |pkg_report, h|
+                report = JSON.parse(File.read(path))
+                report['build_report']['packages']
+                    .each_with_object({}) do |pkg_report, h|
                         h[pkg_report['name']] = pkg_report['built']
                     end
             end
 
+            def load_report(path, root_name, default: { 'packages' => {} })
+                return default unless File.file?(path)
+
+                JSON.parse(File.read(path)).fetch(root_name)
+            end
+
             def consolidated_report
-                cache_pull = File.join(@ws.root_dir, 'cache-pull.json')
-                cache_report = if File.file?(cache_pull)
-                    JSON.load(File.read(cache_pull))
-                else
-                    {}
-                end
+                # NOTE: keys must match PHASES
+                new_reports = {
+                    'import' => @ws.import_report_path,
+                    'build' => @ws.build_report_path,
+                    'test' => @ws.utility_report_path('test')
+                }
+
+                # We start with the cached info (if any) and override with
+                # information from the other phase reports
+                cache_report_path = File.join(@ws.root_dir, 'cache-pull.json')
+                result = load_report(cache_report_path, 'cache_pull_report')['packages']
+                result.delete_if do |_name, info|
+                    next(true) unless info.delete('cached')
```

Comment:
`next true`

Reply:
Mmmmm....

I have to admit that I don't enforce parentheses vs. not parentheses very consistenly.

Since we're going for a consistent style, I would go for "parentheses in imperative calls, non-parentheses in declarative ones unless needed"

E.g.

Good:

~~~
require 'test'
assert_equal 10, this.call
next(true)
FileUtils.mv(something, something)
~~~

Rubocop has a cop for that, but it ignores next unfortunately. Thoughts ?

Reply:
next(true), return(true), break(true) all look weird and unnecessary for me (which is probably why the cop ignores them). To be honest, I don't think I've ever seen this before in any of your code so it will probably take a long time until we have that consistently throughout rock's code base.

Reply:
I think I use a paren quite consistently for `break` and `next` (just checked roby for instance), possibly because I am very unused to have an argument to either of them (it's pretty rare).

I don't mind either way, to be honest. The following looks difficult to read for me:

~~~
next true if something
~~~

while I have no qualms with

~~~
if something
   next true
end
~~~

Reply:
I don't really have a problem with `next(true)` either but with `return(true)` I kinda do and I can't think of a reason to enforce `next(true)` and `return true` (the only real difference between them is the scope). That said, consistency should be the priority so let's stick to what is more common in rock's codebase. Since no one is more familiar with the codebase than you, I will gladly follow your lead on this one

Reply:
OK, since I myself use raise and return without parens even in modifier form, I'd go for going consistent and removing the parens for next and break too.